src/simulators/extended_stabilizer/extended_stabilizer_state.hpp

```diff
@@ -694,6 +695,16 @@ void State::apply_gate(const Operations::Op &op, RngEngine &rng, uint_t rank) {
   case Gates::pauli:
     apply_pauli(op.qubits, op.string_params[0], rank);
     break;
+  case Gates::ecr:
+    BaseState::qreg_.apply_h(op.qubits[1], rank);
+    BaseState::qreg_.apply_s(op.qubits[0], rank);
+    BaseState::qreg_.apply_z(op.qubits[1], rank); // sdg(1)
+    BaseState::qreg_.apply_s(op.qubits[1], rank); // sdg(1)
+    BaseState::qreg_.apply_h(op.qubits[1], rank);
+    BaseState::qreg_.apply_cx(op.qubits[0], op.qubits[1], rank);
+    BaseState::qreg_.apply_x(op.qubits[0], rank);
+    BaseState::qreg_.apply_x(op.qubits[1], rank);
```

Comment:
This implementation differs to ecr by a -pi/4 global phase. is it generally ok ?

Reply:
If I am not mistaken this one doesn't have phase difference 

qc.h(1)
qc.cx(0, 1)
qc.t(1)
qc.cx(0, 1)
qc.h(1)
qc.x(0)
qc.h(1)
qc.cx(0, 1)
qc.tdg(1)
qc.cx(0, 1)
qc.h(1)

Reply:
I coped from stabilizer

https://github.com/Qiskit/qiskit-aer/blob/b706ee1c7445f13f11a28eff4e11a006c89a78dd/src/simulators/stabilizer/stabilizer_state.hpp#L389-L397

But it seems it was not correct, ad I get correct circuit by using transpiler

```
from qiskit import transpile
from qiskit import QuantumCircuit
from qiskit_aer import AerSimulator
sim = AerSimulator(basis_gates = ["h", "cx", "x", "s", "z"])
qc = QuantumCircuit(2)
qc.ecr(0, 1)
qc = transpile(qc,backend=sim,optimization_level=0)
qc.draw('mpl')
```
![image](https://github.com/Qiskit/qiskit-aer/assets/30102488/de636d05-b186-416e-a1ba-48325b797ba6)

I will fix by using this circuit